libs/s25main/desktops/dskCredits.cpp

```diff
@@ -34,21 +35,33 @@
 #include <cstdlib>
 #include <numeric>
 
-/** @class dskCredits
- *
- *  Class for the Credits Desktop.
- */
-
 /// Duration for one credits page
-const unsigned PAGE_TIME = 12900;
+const auto PAGE_TIME = std::chrono::milliseconds(12900);
 /// Duration for fading between pages
-const unsigned FADING_TIME = 2000;
+const auto FADING_TIME = std::chrono::seconds(2);
+using std::chrono::duration_cast;
+
+namespace {
+enum
+{
+    ID_btBack,
+    ID_txtRttr,
+    ID_txtCredits,
+    ID_tmrBobAnim,
+    ID_tmrPage,
+};
+}
 
 dskCredits::dskCredits() : Desktop(LOADER.GetImageN("setup013", 0)), itCurEntry(entries.end())
 {
-    AddTextButton(0, DrawPoint(300, 550), Extent(200, 22), TC_RED1, _("Back"), NormalFont);
-    AddText(1, DrawPoint(400, 10), _("Return To The Roots"), COLOR_YELLOW, FontStyle::CENTER, LargeFont);
-    AddText(2, DrawPoint(400, 33), _("Credits"), COLOR_YELLOW, FontStyle::CENTER, LargeFont);
+    AddTextButton(ID_btBack, DrawPoint(300, 550), Extent(200, 22), TC_RED1, _("Back"), NormalFont);
+    AddText(ID_txtRttr, DrawPoint(400, 10), _("Return To The Roots"), COLOR_YELLOW, FontStyle::CENTER, LargeFont);
+    AddText(ID_txtCredits, DrawPoint(400, 33), _("Credits"), COLOR_YELLOW, FontStyle::CENTER, LargeFont);
+    AddTimer(ID_tmrBobAnim, 40); // Bob animation every 40ms
+    pageTimer = AddTimer(ID_tmrPage, duration_cast<std::chrono::milliseconds>(PAGE_TIME).count());
+    pageTimer->Stop();
+
+    LOADER.Load(ResourceId("credits")); // Ignore failure
```

Comment:
why do we load it twice?

Reply:
Failed attempt I forgot to remove

Reply:
should we at least log here that the resource failed to load? or is it automatically logged?

Reply:
It is. There will be a status with timing on success or fail on error